sickbeard/scene_exceptions.py

```diff
@@ -1,341 +1,350 @@
 
 
 def get_scene_exception_by_name(show_name):
+    """Given a show name, return the first indexerid and season of the exceptions list."""
     return get_scene_exception_by_name_multiple(show_name)[0]
 
 
 def get_scene_exception_by_name_multiple(show_name):
-    """
-    Given a show name, return the indexerid of the exception, None if no exception
-    is present.
-    """
-
-    # try the obvious case first
+    """Given a show name, return the indexerid of the exception, None if no exception is present."""
+    # Try the obvious case first
     cache_db_con = db.DBConnection('cache.db')
     exception_result = cache_db_con.select(
-        'SELECT indexer_id, season FROM scene_exceptions WHERE LOWER(show_name) = ? ORDER BY season ASC',
+        b'SELECT indexer_id, season FROM scene_exceptions WHERE LOWER(show_name) = ? ORDER BY season ASC',
         [show_name.lower()])
     if exception_result:
-        return [(int(x['indexer_id']), int(x['season'])) for x in exception_result]
+        return [(int(x[b'indexer_id']), int(x[b'season'])) for x in exception_result]
 
     out = []
-    all_exception_results = cache_db_con.select('SELECT show_name, indexer_id, season FROM scene_exceptions')
+    all_exception_results = cache_db_con.select(b'SELECT show_name, indexer_id, season FROM scene_exceptions')
 
     for cur_exception in all_exception_results:
 
-        cur_exception_name = cur_exception['show_name']
-        cur_indexer_id = int(cur_exception['indexer_id'])
+        cur_exception_name = cur_exception[b'show_name']
+        cur_indexer_id = int(cur_exception[b'indexer_id'])
 
-        if show_name.lower() in (
-                cur_exception_name.lower(),
-                sickbeard.helpers.sanitizeSceneName(cur_exception_name).lower().replace('.', ' ')):
+        if show_name.lower() in (cur_exception_name.lower(),
+                                 helpers.sanitizeSceneName(cur_exception_name).lower().replace('.', ' ')):
 
-            logger.log(u'Scene exception lookup got indexer id {0}, using that'.format
+            logger.log('Scene exception lookup got indexer ID {0}, using that'.format
                        (cur_indexer_id), logger.DEBUG)
 
-            out.append((cur_indexer_id, int(cur_exception['season'])))
+            out.append((cur_indexer_id, int(cur_exception[b'season'])))
 
     if out:
         return out
 
     return [(None, None)]
 
 
-def retrieve_exceptions():  # pylint:disable=too-many-locals, too-many-branches
-    """
-    Looks up the exceptions on github, parses them into a dict, and inserts them into the
-    scene_exceptions table in cache.db. Also clears the scene name cache.
-    """
+def update_scene_exceptions(indexer_id, scene_exceptions, season=-1):
+    """Given a indexer_id, and a list of all show scene exceptions, update the db."""
+    cache_db_con = db.DBConnection('cache.db')
+    cache_db_con.action(b'DELETE FROM scene_exceptions WHERE indexer_id=? and season=?', [indexer_id, season])
```

Comment:
@medariox seems we already does it here.

so don't understand why we had duplicates